### ISSUE TITLE
custom controls inside of group legends

### ISSUE THREAD

**Issue by Anand-Moghe** [vue-generators/vue-form-generator#451]
I am submitting a feature request.
 
**My Environment:**

  * Version: [2.X.X ]
  * Browser: [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
  * Language:  ES6

I am trying to figure out a way to add my own vue custom control ( right align) inside the **legend** tag rendered by **group schema**.  

But currently the code only renders the text inside the handlebar ( as per the template [here](https://github.com/vue-generators/vue-form-generator/blob/d8436255449b5196c4f652c75ffc55143979f7f4/src/formGenerator.vue#L21)).

Can you suggest a hack or a workaround available for me so that I can insert my own custom control instead of a plain text for a legend? Or any idea how can this feature be implemented ?

Basically I am trying to implement a feature where I duplicate the fields in the schema when I click "Add" button at the right side of the legend. 
E.g 
 
```
this.schema = {
 groups: [   {
       legend: "Preferences", /* This will only add legend text (but I want My own Vue component here) */
        fields: []
     },...
  ]
}
```
 
 
I want to add values to legend so that it can render my own custom control inside <legend></legend>.
Any idea how can I implement this feature/functionality?

**Comment by Anand-Moghe** [comment]
One approach I see is to not use group schema and use only fields schema. Then  use my own custom control to render the legend tag the way I wanted. Because I am not sure if I can change the rendering of the legend component in case of a group schema.

**Comment by zoul0813** [comment]
I’m not sure how involved your form is, but you might consider a custom component for VFG.  The custom component could contain another VFG instance ... for example.

In addition, you can use the VFG instance as a parent and provide a new template, similar to how the components use abstractField.  This would allow you to customize how VFG builds the layout, allowing you to inject your custom legend component.

I’m not sure your request would make it into the core code, but I am interested in how you solve this problem so keep me posted.

**Comment by Anand-Moghe** [comment]
I solved it by using the custom component (as per the documentation) and not using the group schema.

**Comment by zoul0813** [comment]
@Anand, look at the abstractField code and the core components to see how validation is wired up... for the most part, text fields shouldn’t need special rules... but date and number fields will need some, checkboxes I believe have some logic as well.
